### PR TITLE
Feature/overview docs

### PR DESCRIPTION
**Pull request by neewy**
### Benefits

Explanation of performance and difference compared to other platforms.

### Possible Drawbacks 

Performance data are old and should be updated, if possible. Since we are investigating how to use Caliper, it might be done even after release.

### Usage Examples or Tests *[optional]*

Just compile docs if you want to.

### CODE REVIEW

File: docs/source/overview.rst [hunk 1]
```diff
@@ -4,17 +4,60 @@ Overview of Iroha
...
+- Modular design, driven by command–query separation principle
+- Assets and identity management
+
+In our quality model, we focus on and continuously improve:
+
+- Reliability (fault tolerance, recoverability)
+- Performance Efficiency (in particular time-behavior and resource utilization)
+- Usability (learnability, user error protection, appropriateness recognisability)
+
 Where Iroha can be used?
```
Comment (igor-egorov) [inline docs/source/overview.rst]: Where **can Iroha** be used?

Please consider changing the word order.

File: docs/source/overview.rst [hunk 2]
```diff
@@ -4,17 +4,60 @@ Overview of Iroha
...
 ------------------------
 
+Hyperledger Iroha is a general purpose permissioned blockchain system that can be used to manage digital assets, identity, and serialized data.
+This can be useful for applications such as interbank settlement, central bank digital currencies, payment systems, national IDs, and logistics, among other.
```
Comment (igor-egorov) [inline docs/source/overview.rst]: ... among other -> among **others**  ?

File: docs/source/overview.rst [hunk 3]
```diff
@@ -4,17 +4,60 @@ Overview of Iroha
...
+
+For a detailed description please check our `Use Case Scenarios section <http://iroha.readthedocs.io/en/latest/use_cases/>`_.
+
 How is it different from Bitcoin or Ethereum?
 ---------------------------------------------
 
+Bitcoin and Ethereum are designed to be permissionless ledgers where anyone can join and access all the data.
+They also have native cryptocurrencies that are required to interact with the systems.
+
+In Iroha there is no native cryptocurrency. Instead, to meet the needs of enterprises, system interaction is permissioned, meaning that only people with requisite access can interact with the system. Additionally, queries are also permissioned, such that access to all the data can be controlled.
```
Comment (igor-egorov) [inline docs/source/overview.rst]: In Iroha, there is no ...
Consider adding a comma.

File: docs/source/overview.rst [hunk 4]
```diff
@@ -4,17 +4,60 @@ Overview of Iroha
...
+
 Is it fast?
 -----------
 
+As per the latest review date of these docs, according to `Huawei Caliper <https://github.com/hyperledger/caliper>`_ testing tool, Iroha is capable of processing 45 transactions per second. Theoretically, this is not even close to the limit of the system, and we will continue constant optimizations in order to improve stability and performance.
+
 How to create applications around Iroha?
 ----------------------------------------
+
+In order to bring power of blockchain into your application you should think first of how it is going to interface with Iroha peers.
```
Comment (igor-egorov) [inline docs/source/overview.rst]: Probably **the** article is missing: In order to bring **the** power of blockchain ...

Comment (igor-egorov) [reply docs/source/overview.rst]: Please consider adding a comma after "application":
In order to bring ... into your application, you should think ...

File: docs/source/overview.rst [hunk 5]
```diff
@@ -4,17 +4,60 @@ Overview of Iroha
...
+
+
+One major difference from Ethereum, in particular, is that Hyperledger Iroha allows users to perform common functions, such as creating and transferring digital assets, by using prebuilt commands that are in the system.
+This negates the need to write cumbersome and hard to test smart contracts, enabling developers to complete simple tasks faster and with less risk.
+
 How is it different from the rest of Hyperledger frameworks or other permissioned blockchains?
 ----------------------------------------------------------------------------------------------
 
+Iroha has a novel, Byzantine fault tolerant consensus algorithm (called YAC [#f1]_) that is high-performance and allows for finality of transactions with low latency.
+Other frameworks either focus more on probabilistic consensus algorithms, such as Nakamoto Consensus, or sacrifice Byzantine fault tolerance.
```
Comment (grimadas) [inline docs/source/overview.rst]: Nakamoto, sacrificing transaction finality

File: docs/source/overview.rst [hunk 6]
```diff
@@ -4,17 +4,60 @@ Overview of Iroha
...
+As per the latest review date of these docs, according to `Huawei Caliper <https://github.com/hyperledger/caliper>`_ testing tool, Iroha is capable of processing 45 transactions per second. Theoretically, this is not even close to the limit of the system, and we will continue constant optimizations in order to improve stability and performance.
+
 How to create applications around Iroha?
 ----------------------------------------
+
+In order to bring the power of blockchain into your application, you should think first of how it is going to interface with Iroha peers.
+A good start is to check `Core Concepts section <http://iroha.readthedocs.io/en/latest/core_concepts/>`_, explaining what exactly is a transaction and query, and how users of your application are supposed to interact with it.
+
+We also have several client libraries which provide tools for developers to form building blocks, such as signatures, commands,
+send messages to Iroha peers and check the status.
```
Comment (grimadas) [inline docs/source/overview.rst]: Is there a redundant enter?

File: docs/source/overview.rst [hunk 7]
```diff
@@ -4,17 +4,60 @@ Overview of Iroha
...
+One major difference from Ethereum, in particular, is that Hyperledger Iroha allows users to perform common functions, such as creating and transferring digital assets, by using prebuilt commands that are in the system.
+This negates the need to write cumbersome and hard to test smart contracts, enabling developers to complete simple tasks faster and with less risk.
+
 How is it different from the rest of Hyperledger frameworks or other permissioned blockchains?
 ----------------------------------------------------------------------------------------------
 
+Iroha has a novel, Byzantine fault tolerant consensus algorithm (called YAC [#f1]_) that is high-performance and allows for finality of transactions with low latency.
+Other frameworks either focus more on probabilistic consensus algorithms, such as Nakamoto Consensus, or sacrifice Byzantine fault tolerance.
+
+Also, Iroha's built-in commands are a major benefit compared to other platforms, since it is very simple to do common tasks such as create digital assets, register accounts, and transfer assets between accounts.
```
Comment (grimadas) [inline docs/source/overview.rst]: Also you can mention that is narrows the attack vector, improving overall security of the system, as there are less things to fail

File: docs/source/overview.rst (hunk 5 above)
Comment (neewy) [reply docs/source/overview.rst]: either focus or sacrifice

Comment (neewy) [reply docs/source/overview.rst]: Nakamoto Consensus is a widespread term :)

File: docs/source/overview.rst (hunk 6 above)
Comment (neewy) [reply docs/source/overview.rst]: No, separate paragraphs

**Pull request merged**